hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/client/OzoneQuota.java

```diff
@@ -190,14 +201,44 @@ public static OzoneQuota parseQuota(String quotaInBytes,
           " value cannot be greater than Long.MAX_VALUE BYTES");
     }
 
-    if (nSize < 0) {
-      throw new IllegalArgumentException("Quota cannot be negative.");
+    if (nSize <= 0) {
+      throw new IllegalArgumentException("Invalid values for quota: " + nSize);
     }
 
-    return new OzoneQuota(quotaInNamespace,
-        new RawQuotaInBytes(currUnit, nSize));
+    return new OzoneQuota(new RawQuotaInBytes(currUnit, nSize));
   }
 
+  /**
+   * Parses a user provided string Namespace quota and returns the
+   * Quota Object.
+   *
+   * @param quotaInNamespace Volume quota in counts
+   *
+   * @return OzoneQuota object
+   */
+  public static OzoneQuota parseNameSpaceQuota(String quotaInNamespace) {
+    long nameSpaceQuota = Long.parseLong(quotaInNamespace);
+    if (nameSpaceQuota <= 0) {
+      throw new IllegalArgumentException(
+          "Invalid values for quota: " + nameSpaceQuota);
+    }
+    return new OzoneQuota(nameSpaceQuota, new RawQuotaInBytes(Units.BYTES, -1));
+  }
+
+  /**
+   * Parses a user provided string and returns the
+   * Quota Object.
+   *
+   * @param quotaInBytes Volume quota in bytes
+   * @param quotaInNamespace Volume quota in counts
+   *
+   * @return OzoneQuota object
+   */
+  public static OzoneQuota parseQuota(String quotaInBytes,
+      String quotaInNamespace) {
```

Comment:
@captainzmc ， can we rollback long to string convert for namspace quota now? As default quota 0 will all be converted to Quota.RESET based on above approach that we discussed.

Reply:
The above advice only solves the problem of creating a Volume or Bucket.
The string quotainNamespace in parseQuota is still useful for verifying whether the user passed quotainNamespace.
Let's look at this case:
If the original volume space and namespace quoat were 100 and 10T, respectively, we reset the space quota separately.
ozone sh volume setquota vol1 --space-quota 20TB
OzoneConsts.QUOTA_RESET is the default quota for VolumeArgs. The space and namespace quota for vol1 will change to -1 and 20T, which is not expected.

I think it makes more sense to handle these cases in Ozonequota #parseQuota. Also, spaceQuota is currently a String in parseQuota, which makes our client-side processing logic more consistent.

Reply:
Hi @linyiqun , thanks for review the code.  We found bugs in production cluster with the previous implementation which doesn't distinguish the cases that user doesn't set the quota and user set the quota to 0.  So it's better to make it clear to avoid bugs caused by concept vague.  

In summary, for quota in bucket and volume,  
1) quota value will be -2  to indicates it's 1.0.0 bucket/volume 
2) quota value will be -1 if  bucket/volume created without explicitly set quota value in 1.1.0
3) quota > 0 indicates quota is enabled
4) throw exception if user set quota to 0 to keep consistent with current HDFS behavior

Reply:
Comments make sense to me, @captainzmc , @ChenSammi